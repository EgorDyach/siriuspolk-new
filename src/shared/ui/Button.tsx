import { cx } from 'class-variance-authority';
import React, { ComponentProps, ReactNode } from 'react';

interface ButtonProps extends ComponentProps<'button'> {
  children?: ReactNode;
}

function Button({ className, children, ...props }: ButtonProps) {
  return (
    <>
      <button
        className={cx(
          'py-[12px] px-[32px] text-white text-2xl font-lora bg-[#343434] cursor-pointer',
          'aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive',
          className,
        )}
        {...props}
      >
        {children}
      </button>
    </>
  );
}

export { Button };
