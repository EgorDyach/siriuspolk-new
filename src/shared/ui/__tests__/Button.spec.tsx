import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from '../Button';

describe('<Button />', () => {
  it('should render button correctly', () => {
    render(<Button />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('should correctly render with classname', () => {
    render(<Button className="test-class" />);
    const button = screen.getByRole('button');
    expect(button).toHaveClass('test-class');
    expect(button).toHaveClass('py-[16px]', 'px-[40px]', 'text-white');
  });

  it('should correctly render with small size', () => {
    render(<Button size="small" />);
    const button = screen.getByRole('button');
    expect(button).toHaveClass(
      'text-sm',
      'md:text-base',
      'py-[12px]',
      'px-[32px]',
    );
  });

  it('should correctly render with gray theme', () => {
    render(<Button theme="gray" />);
    const button = screen.getByRole('button');
    expect(button).toHaveClass(
      'bg-[#D9D9D9]',
      '!text-black',
      '[&>*]:text-black',
    );
  });

  it('should correctly render with children', () => {
    render(
      <Button>
        <div data-testid="children" />
      </Button>,
    );
    const children = screen.getByTestId('children');
    expect(children).toBeInTheDocument();
    expect(children.parentElement).toHaveRole('button');
  });

  it('should correctly work onClick', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick} />);
    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
  });

  it('should render as disabled when disabled prop is passed', () => {
    render(<Button disabled={true} />);
    const button = screen.getByRole('button');
    expect(button).toBeDisabled();
  });

  it('matches snapshot with default Button', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with children', () => {
    const { asFragment } = render(<Button>Test</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with disabled', () => {
    const { asFragment } = render(<Button disabled />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with disabled', () => {
    const { asFragment } = render(<Button disabled />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('matches snapshot with small size', () => {
    const { asFragment } = render(<Button size="small" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('matches snapshot with gray theme', () => {
    const { asFragment } = render(<Button theme="gray" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
