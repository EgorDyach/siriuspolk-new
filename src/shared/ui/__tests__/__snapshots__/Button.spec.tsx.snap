// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot with children 1`] = `
<DocumentFragment>
  <button
    class="py-[16px] px-[40px] text-base md:text-2xl font-lora bg-[#343434] cursor-pointer aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-white"
  >
    Test
  </button>
</DocumentFragment>
`;

exports[`<Button /> matches snapshot with default Button 1`] = `
<DocumentFragment>
  <button
    class="py-[16px] px-[40px] text-base md:text-2xl font-lora bg-[#343434] cursor-pointer aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-white"
  />
</DocumentFragment>
`;

exports[`<Button /> matches snapshot with disabled 1`] = `
<DocumentFragment>
  <button
    class="py-[16px] px-[40px] text-base md:text-2xl font-lora bg-[#343434] cursor-pointer aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-white"
    disabled=""
  />
</DocumentFragment>
`;

exports[`<Button /> matches snapshot with disabled 2`] = `
<DocumentFragment>
  <button
    class="py-[16px] px-[40px] text-base md:text-2xl font-lora bg-[#343434] cursor-pointer aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-white"
    disabled=""
  />
</DocumentFragment>
`;

exports[`<Button /> matches snapshot with gray theme 1`] = `
<DocumentFragment>
  <button
    class="bg-[#D9D9D9] !text-black [&>*]:text-black py-[16px] px-[40px] text-base md:text-2xl font-lora bg-[#343434] cursor-pointer aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-white"
  />
</DocumentFragment>
`;

exports[`<Button /> matches snapshot with small size 1`] = `
<DocumentFragment>
  <button
    class="text-sm md:text-base py-[12px] px-[32px] py-[16px] px-[40px] text-base md:text-2xl font-lora bg-[#343434] cursor-pointer aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-white"
  />
</DocumentFragment>
`;
